gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-default-page/api-documentation-v4-default-page.component.ts

```diff
@@ -48,26 +50,15 @@ export class ApiDocumentationV4DefaultPageComponent implements OnInit, OnDestroy
   ) {}
 
   ngOnInit() {
-    this.isLoading = true;
-
-    this.apiV2Service
-      .get(this.activatedRoute.snapshot.params.apiId)
-      .pipe(
-        tap((api) => {
-          this.api = api;
-          this.isReadOnly = api.originContext?.origin === 'KUBERNETES';
-        }),
-        switchMap(() => this.activatedRoute.queryParams),
-        switchMap(() => {
-          return this.apiDocumentationV2Service.getApiPages(this.api.id);
-        }),
-        takeUntil(this.unsubscribe$),
-      )
-      .subscribe((res) => {
-        this.pages = res.pages;
-        this.homepage = this.pages.find((p) => p.homepage === true);
-        this.isLoading = false;
-      });
+    this.data$ = this.activatedRoute.params.pipe(
+      switchMap(({ apiId }) => (apiId ? combineLatest([this.apiV2Service.get(apiId), this.getApiPages(apiId)]) : EMPTY)),
+      map(([api, pagesResponse]) => {
+        this.api = api;
+        this.isReadOnly = api.originContext?.origin === 'KUBERNETES';
+        return { hasCustomPages: pagesResponse.pages?.length > 0, homepage: pagesResponse.pages.find((p) => p.homepage === true) };
```

Comment:
If you wish you can break down the return statement like this:
`      const homepage = pagesResponse.pages.find(p => p.homepage);
      return { hasCustomPages: !!pagesResponse.pages.length, homepage };`

Reply:
Done